buildkite/bazelci.py

```diff
@@ -580,25 +590,108 @@ def _get_buildkite_token(self):
             else "buildkite-untrusted-api-token",
         )
 
-    def _open_url(self, url):
+
+    def _open_url(self, url, params = None):
         try:
+            params_str = ""
+            if params:
+                for k, v in params:
+                    params_str = params_str + "&{}={}".format(k, v)
             return (
-                urllib.request.urlopen("{}?access_token={}".format(url, self._token))
+                urllib.request.urlopen("{}?access_token={}{}".format(url, self._token, params_str))
                 .read()
                 .decode("utf-8")
             )
         except urllib.error.HTTPError as ex:
             raise BuildkiteException("Failed to open {}: {} - {}".format(url, ex.code, ex.reason))
 
+
+    # Get build info for a pipeline with a given build number
```

Comment:
Here and below: can you add the text as pydoc comments instead of "real" comments?